src/editor/Editor.js

```diff
@@ -435,6 +433,9 @@ define(function (require, exports, module) {
 
         this.on("cursorActivity", function (event, editor) {
             self._handleCursorActivity(event);
+            if(MainViewManager._findProvider(editor)){
+                self.trigger("providerAvailable", self);
```

Comment:
I wonder if we can also send the type of provider when we do the event.  It would be useful for showing different icons in the UI depending on what's available.